Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -0,0 +1,247 @@
+
+		Expected Return:
+		<testsuites failures="1" id="20191011_175324" tests="1">
+			<testsuite failures="1" name="Example Tests" tests="1">
+				<testcase name="Test1">
+					<failure type="failure">
+						Expected: add_nums(1, 1) equal to 3
+						But: was 2
+					</failure>
+				</testcase>
+			</testsuite>
+		</testsuites>
+        ---------
+*/
+	show_xml_report = Method({xml_results},
+		start_line = "<?xml version=\!"1.0\!" encoding=\!"UTF-8\!" ?>\!n";
+		
+		content = "";
+		test_suites = xml_results << get keys;
+		test_cases_set = xml_results << get values;
+		For(i=1, i<=NItems(test_suites), i++,
+		
+
+			test_cases = test_cases_set[i];
+			test_case_names = test_cases << get keys;
+			test_case_info = test_cases << get values;
+			test_case_xml = "";
+			For(k=1, k<=NItems(test_case_names), k++,
+
+				info_types = test_case_info[k] << get keys;
+				
+				test_case_content = "";
+				For(l=1, l<=NItems(test_case_info[k]), l++,
+					info_type = info_types[l];
+					props = Associative Array();
+					props["type"] = info_type;
+					failure_content = this:generate_xml_tag("failure", test_case_info[k][info_type], props);
+
+					test_case_content ||= failure_content;
+
+				);
+				props = Associative Array();
+				props["name"] = test_case_names[k];
+				test_case_xml ||= this:generate_xml_tag("testcase", trim(test_case_content), props);
+				
+			);
+			num_tests = NItems(test_case_names);
+			props = Associative Array();
+			props["name"] = test_suites[i];
+			props["tests"] = Char(num_tests);
+
+			// find test results that are non empty
+			num_successes = Nitems(loc(test_case_info, [=>]));
+			props["failures"] = Char(num_tests-num_successes);
+			test_suite_tag = this:generate_xml_tag("testsuite", trim(test_case_xml), props);
+			content ||= test_suite_tag;
+		);
+		
+		props = Associative Array();
+		date = today();
+		props["id"] = format(date, "yyyymmdd" ) || "_" || Char(Hour(date)) || Char(Minute(date)) || Char(Second(date));
+		props["tests"] = Char(Nitems(this:successes) + NItems(this:failures) + NItems(this:unexpected throws));
+		props["failures"] = Char(NItems(this:failures) + NItems(this:unexpected throws));
+
+		xml_result = start_line || this:generate_xml_tag("testsuites", trim(content), props);;
+		Return(xml_result);
+	);
+	
+	// Function: show report
+	//
+	// Dump a report to the log detailing all the successes and
+	// failures for each test in JunitXML format shown in the comments at the top.
+	show report = Method( {},
+		num successes = Length( this:successes );
+		num failures = Length( this:failures );
+		num unexpected throws = Length( this:unexpected throws );
+
+		hash_results = Associative Array();
+
+		For( i = 1, i <= num successes, i++,
+			char test expr = If( Is Expr( Name Expr( this:successes[i][2] ) ),
+				Char( Name Expr( this:successes[i][2] ) ),
+				Char( this:successes[i][2] )
+			) || " ";
+			tmp_label = this:successes[i][1];
+			list_tmp_label = words(tmp_label, "⮚");
```

Comment:
This should be [ut concat test label sep](https://github.com/sassoftware/jsl-hamcrest/blob/5a0d45b6341a947056edd250c3484be18cf77b5f/Source/Core/Utils.jsl#L466-L467) everywhere.

Reply:
okay, I will make that change